lib/objectid.js

```diff
@@ -0,0 +1,12 @@
+var ObjectId;
+
+try {
+  ObjectId = require('mongodb').ObjectId;
+} catch(e) {
+  if (e.code === 'MODULE_NOT_FOUND')
+    throw e
+}
+
```

Comment:
>Makes me wonder if there is a way how to allow loopback-connector-mongodb to call modelTypes.registerType(ObjectId) during connector/datasource initialisation. That way we can keep the implementation in mongodb and still have ObjectID available in the app.

I have thought about this too, but figured it would be harder to implement because we don't expose that part of the lib in DAO. The long term solution I had in mind and discussed with @ritch is an app-level registry for datatypes. During boot time, we would register all basic built in datatypes, then register any custom types. Then at run time, we can dynamically register/deregister new datatypes. Something along the lines of:

`app.datatypes.add(ObjectId);
`app.datatypes.list();
`app.datatypes.remove(ObjectId);

Or use an a K/V array for datatypes, etc. We can discuss further in the long-term proposal with regards to implementations.